corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -401,10 +401,14 @@ def hard_delete_forms(domain, form_ids, delete_attachments=True):
         if delete_attachments:
             attachments = list(FormAccessorSQL.get_attachments_for_forms(form_ids))
 
-        with get_cursor(XFormInstanceSQL) as cursor:
-            cursor.execute('SELECT hard_delete_forms(%s, %s) AS deleted_count', [domain, form_ids])
-            results = fetchall_as_namedtuple(cursor)
-            deleted_count = sum([result.deleted_count for result in results])
+        db_form_ids = split_list_by_db_partition(form_ids)
+        deleted_count = 0
+        for db_name, form_ids in db_form_ids.items():
+            # cascade should delete the attachments and operations
+            _, deleted_models = XFormInstanceSQL.objects.using(db_name).filter(
+                domain=domain, form_id__in=form_ids
+            ).delete()
+            deleted_count += deleted_models.get(XFormInstanceSQL._meta.label, 0)
```

Comment:
Also moved this to Django which was the only other non-django query happening `save_processed_models`